metadata-builders/src/main/java/com/linkedin/metadata/builders/search/DatasetIndexBuilder.java

```diff
@@ -107,6 +109,22 @@ private DatasetDocument getDocumentToUpdateFromAspect(@Nonnull DatasetUrn urn,
             .collect(Collectors.toList())));
   }
 
+  @Nonnull
+  private DatasetDocument getDocumentToUpdateFromAspect(@Nonnull DatasetUrn urn,
+                                                        @Nonnull GlossaryTerms glossaryTerms) {
+    return new DatasetDocument().setUrn(urn)
+            .setGlossaryTerms(new StringArray(glossaryTerms.getTerms()
+                    .stream()
+                    .map(term -> {
+                      String name = term.getUrn().getNameEntity();
+                      if (name.contains(".")) {
+                        String[] nodes = name.split(Pattern.quote("."));
+                        return nodes[nodes.length - 1];
```

Comment:
why do we only want to index the last element of the Urn?

Reply:
Example: glossary term name(with hierarchy) comes out `instruments.FinancialInstrument_v1` here we just want to index `FinancialInstrument_v1` because this is our glossary term name.

Reply:
If someone searches for `instruments` do you not want any hits on this entity or entity-field?

Reply:
no. when a user searches for `FinancialInstrument_v1` then I should be able to get the linked dataset. Because `FinancialInstrument_v1` is my glossary term name. User shouldn't care about the domain of the glossary term.